src/pydis_jam23/ui_app.py

```diff
@@ -0,0 +1,147 @@
+        self._temp()
+        self._functions()
+        self.statusBar.showMessage("Application Started")
+        self.show()  # displays the QMainWindow in QApplication Instance
+
+    def _temp(self):
+        self._image_tmp = None
+        self.image = None
+
+    def _functions(self):
+        # initial widget state configs
+        self.lsbframe.hide()
+        self.lsbinsertdata.hide()
+
+        # shortcut key actions
+        self.action_open_image.setShortcut("Ctrl+O")
+        self.action_save_image.setShortcut("Ctrl+S")
+        self.action_exit_app.setShortcut("Ctrl+Q")
+
+        # lambda functions triggered by ui elements/buttons
+        self.action_open_image.triggered.connect(lambda: AppFuncs.action_open_image(self))
+        self.action_save_image.triggered.connect(lambda: AppFuncs.action_save_image(self))
+        self.action_exit_app.triggered.connect(lambda: AppFuncs.action_exit_app(self))
+        self.lsbtoolbtn.clicked.connect(lambda: AppFuncs.lsbtoolsection(self.statusBar, self.lsbframe))
+        self.lsbaction.currentTextChanged.connect(lambda: AppFuncs.lsbaction(self))
+        # self.lsbcodec.currentTextChanged.connect(lambda: AppFuncs.lsbsetcodec(self) )
+        self.lsbinsertbtn.clicked.connect(lambda: AppFuncs.lsbinsertdata(self))
+        self.lsbreceivebtn.clicked.connect(lambda: AppFuncs.lsbreceivedata(self))
+
+    # def resizeEvent(self, event):
+    # ^ this is a problem for hatch formatting because this method is inherited from QMainWindow
+    #     # self.image_scaled = image.scaled(self.scroll.width(),self.scroll.height())
+    #     # self.pixmap = QPixmap.fromImage(self.image_scaled)
+    #     # self.imageLabel.setPixmap(self.pixmap)
+    #     QtWidgets.QMainWindow.resizeEvent(self, event)
+    #     print(event)
+
+
+class AppFuncs:
+    @staticmethod
+    def action_open_image(root):
+        # this function opens a image in the gui app
+        root.statusBar.showMessage("file dialog opened")
+        filename = QtWidgets.QFileDialog.getOpenFileName()  # This is temporary
+        # print(filename)
+        root.statusBar.showMessage(f"Image: {filename[0]}")
+        # dlg = QtWidgets.QFileDialog()
+        # dlg.setFileMode(QtWidgets.QFileDialog.AnyFile)
+        # dlg.setFilter("Text files (*.*)")
+        # filenames = QtCore.QStringListModel()
+        # print(filenames)
+        # image_viewer to display opened image
+        if filename[0] != "":
+            root.image_viewer.setPixmap(QtGui.QPixmap(filename[0]))
+            root.statusBar.showMessage(f"Displaying: {filename[0]}")
+            # need to find a zooming solution for the image
+            root.image = filename[0]
+            root._image_tmp = Image.open(root.image)
+
+    @staticmethod
+    def action_save_image(root):
+        # this function saves a image in the gui app
+        print("saving image")
+        # with open('newImage.jpg', 'wb') as imagefile:
+        #     imagefile.write(root._image_tmp)
```

Comment:
Commented code.

Also, debugging print statement, but maybe this is ok/beneficial in a GUI app? Not sure best practices here. Would probably be better to use the `logging` module, but this is fine for now.